facia/app/controllers/FaciaController.scala

```diff
@@ -171,6 +171,26 @@ trait FaciaController extends Controller with Logging with ExecutionContexts wit
       }
   }
 
+  def renderShowMore(path: String, collectionId: String) = MemcachedAction { implicit request =>
+    for {
+      maybeFaciaPage <- frontJson.get(path)
+    } yield {
+      val maybeResponse = for {
+        faciaPage <- maybeFaciaPage
+        (container, index) <- faciaPage.front.containers.zipWithIndex.find(_._1.dataId == collectionId)
```

Comment:
is there a future optimisation in there to not parse the whole json just to find one collection?